share/ipld/nmt_adder.go

```diff
@@ -12,6 +14,7 @@ import (
 // NmtNodeAdder adds ipld.Nodes to the underlying ipld.Batch if it is inserted
 // into a nmt tree.
 type NmtNodeAdder struct {
+	lock   sync.Mutex
```

Comment:
Let's document that the lock is used for both the Batch and cid.Set

Reply:
It is used for whole `NmtNodeAdder`. Do you think its usage could be confusing?

Reply:
Just a quick mention that cover all the fields wouldn't hurt